Mathlib/Data/List/Basic.lean

```diff
@@ -207,6 +207,8 @@ theorem append_left_injective (t : List α) : Injective fun s ↦ s ++ t :=
 
 /-! ### replicate -/
 
+attribute [simp] map_const'
+
```

Comment:
I'm not convinced this is the obvious location. I suppose the RHS *is* about `replicate`. But based on the name and the LHS I would expect it lower down in the `map` section.

Reply:
Could you also add a comment explaining the motivation? (Which can be useful if we ever want to move this change into core Lean.)

Reply:
My reasoning was that `List.map` is defined earlier than `List.replicate` in the Core file. But I guess that really doesn't matter